components/org.wso2.carbon.identity.tenant.resource.manager/src/main/java/org/wso2/carbon/identity/tenant/resource/manager/TenantAwareAxis2ConfigurationContextObserver.java

```diff
@@ -0,0 +1,188 @@
+import org.wso2.carbon.utils.AbstractAxis2ConfigurationContextObserver;
+import org.wso2.carbon.utils.multitenancy.MultitenantConstants;
+import java.util.List;
+
+import static org.wso2.carbon.identity.tenant.resource.manager.constants.TenantResourceConstants.ErrorMessages.ERROR_CODE_ERROR_WHEN_CREATING_TENANT_EVENT_PUBLISHER_CONFIGURATION_BY_CONFIG_STORE;
+import static org.wso2.carbon.identity.tenant.resource.manager.constants.TenantResourceConstants.ErrorMessages.ERROR_CODE_ERROR_WHEN_CREATING_TENANT_EVENT_PUBLISHER_CONFIGURATION_USING_SUPER_TENANT_CONFIG;
+import static org.wso2.carbon.identity.tenant.resource.manager.util.ResourceUtils.getResourceFile;
+import static org.wso2.carbon.identity.tenant.resource.manager.util.ResourceUtils.getResourceManager;
+import static org.wso2.carbon.identity.tenant.resource.manager.util.ResourceUtils.handleServerException;
+
+/**
+ * Axis2Observer for generating tenant wise publisher configurations.
+ */
+public class TenantAwareAxis2ConfigurationContextObserver extends AbstractAxis2ConfigurationContextObserver {
+
+    private static final Log log = LogFactory.getLog(TenantAwareAxis2ConfigurationContextObserver.class);
+
+    /**
+     * Add the tenant wise publisher and stream Configuration in tenant loading.
+     *
+     * @param tenantId tenant ID.
+     */
+    @Override
+    public void creatingConfigurationContext(int tenantId) {
+
+        log.info("creating configuration context for tenant id: " + tenantId);
+        String tenantDomain = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+        List<EventPublisherConfiguration> activeEventPublisherConfigurations;
+        List<EventStreamConfiguration> eventStreamConfigurationList;
+
+        startSuperTenantFlow();
+
+        activeEventPublisherConfigurations = getSuperTenantEventPublisherConfigurations(tenantId);
+        eventStreamConfigurationList = getSuperTenantEventStreamConfigurations(tenantId);
+
+        startTenantFlow(tenantId, tenantDomain);
+
+        createTenantEventStreamConfiguration(tenantId, eventStreamConfigurationList);
+
+        if (activeEventPublisherConfigurations != null) {
+            createTenantEventPublisherConfiguration(tenantId, activeEventPublisherConfigurations);
+        }
+    }
+
+    private List<EventPublisherConfiguration> getSuperTenantEventPublisherConfigurations(int tenantId) {
+
+        List<EventPublisherConfiguration> activeEventPublisherConfigurations = null;
+        try {
+            activeEventPublisherConfigurations = TenantResourceManagerDataHolder.getInstance()
+                    .getEventPublisherService().getAllActiveEventPublisherConfigurations();
+        } catch (EventPublisherConfigurationException e) {
+            log.error(handleServerException(
+                    TenantResourceConstants.ErrorMessages.ERROR_CODE_ERROR_WHEN_FETCHING_SUPER_TENANT_EVENT_PUBLISHER_CONFIGURATION,
+                    e, String.valueOf(tenantId)));
```

Comment:
Isn't tenant domain is more informative that the tenant id. Can we use the tenant domain here

Reply:
done